StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/Embedded/EmbeddedPaymentElement.swift

```diff
@@ -127,15 +127,30 @@ public final class EmbeddedPaymentElement {
                 return UpdateResult.canceled
             }
 
-            // 2. Re-initialize embedded view to update the UI to match the newly loaded data.
+            
+            // 2.1. Re-initialize embedded form view controller to update the UI to match the newly loaded data.
+            var formViewController: EmbeddedFormViewController?
+            if let formPaymentMethodType = self?.formViewController?.selectedPaymentOption?.paymentMethodType {
+                formViewController = EmbeddedFormViewController(configuration: configuration,
+                                                                intent: loadResult.intent,
+                                                                elementsSession: loadResult.elementsSession,
+                                                                savedPaymentMethods: loadResult.savedPaymentMethods,
+                                                                paymentMethodType: formPaymentMethodType,
+                                                                previousPaymentOption: self?.formViewController?.selectedPaymentOption,
+                                                                analyticsHelper: analyticsHelper,
+                                                                formCache: .init()) // Always use a fresh cache after an update
```

Comment:
nit: instead of having formCache a required arg, make the default parameter use a fresh cache?